### PR TITLE
py-fitter: add v1.5.2, v1.6.0

### PR DESCRIPTION
**Pull request by carsonwoods**
Simple PR which just adds the two new versions of the py-fitter package.

### CODE REVIEW

**Review by manuelakuhn — COMMENTED** [review]
Since `py-fitter@1.6.0` the build system changed from `py-setuptools` to `py-poetry` and the dependencies also have version restrictions (see [pyproject.toml@L28-L35](https://github.com/cokelaer/fitter/blob/v1.6.0/pyproject.toml#L28-L35):
python = "^3.8"
click = "^8.1.6"
joblib = "^1.3.1"
matplotlib = ">=3.7.2"
numpy = ">=1.20"
pandas = "^2.0.3"
scipy = ">=0.18"
tqdm = "^4.65.1"

**Comment by alalazo** [comment]
@carsonwoods Can you incorporate @manuelakuhn suggestions, and ping me for a review?

**Comment by carsonwoods** [comment]
@alalazo Thanks for reminding me about this. I've added the suggestions and it seems like its ready for a review.

File: var/spack/repos/builtin/packages/py-fitter/package.py [hunk 1]
```diff
@@ -12,20 +12,23 @@ class PyFitter(PythonPackage):
...
-    depends_on("py-scipy@0.18:", type=("build", "run"))
-    depends_on("py-tqdm", type=("build", "run"))
-    depends_on("py-joblib", type=("build", "run"))
-    depends_on("py-click", type=("build", "run"))
-    depends_on("py-matplotlib", type=("build", "run"))
+    depends_on("py-poetry-core@1:", when="@1.6:", type="build")
+    depends_on("py-setuptools", when="@:1.6", type="build")
+    depends_on("python@3.8:3.9", type=("build", "run"))
+    depends_on("py-click@8.1.6:9", type=("build", "run"))
+    depends_on("py-joblib@1.3.1:2", type=("build", "run"))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-fitter/package.py]: I think, given our version semantic, you mean:
```suggestion
    depends_on("py-joblib@1.3.1:1", type=("build", "run"))
```
i.e. `1.3.1 < x < 2` ? Similarly for other `depends_on` statements.

Comment (carsonwoods) [reply var/spack/repos/builtin/packages/py-fitter/package.py]: Whoops, that's correct. I just forgot the proper semantics. Thanks!

**Review by alalazo — APPROVED** [review]

**Pull request merged**